src/Symfony/Component/Intl/Tests/NumberFormatter/AbstractNumberFormatterTest.php

```diff
@@ -789,13 +780,24 @@ public function parseTypeDoubleProvider()
         );
     }
 
-    public function testParseTypeCurrency()
+    private function handleExpectedWarningException()
```

Comment:
because the test suite is full of "setExpectedException", I think we should remove this method and do the check inline instead (unfortunately...).

the most concise way I can think of is:
`$this->{method_exists($this, $_ = 'expectException') ? $_ : 'setExpectedException'}(...)`

can you update the test suite in the same PR also please?